### ISSUE TITLE
How soon will FireKit support version STU3 of FHIR? 

### ISSUE THREAD

**Issue by qcastellanos** [ryanbaldwin/FireKit#11]
Hi Ryan,

My team would like to leverage FireKit on our project but need to support STU3. I noticed that your README states that "An STU3 version for Swift 3.1 should be coming soon-ish". Is this still the case and if so, when do you think it will be available?

Thanks,
Quyen Castellanos
quyen.castellanos@parivedasolutions.com

**Comment by ryanbaldwin** [comment]
Hi Quyen!

Thanks for your interest. I'm hoping to get on this sooner rather than later, and really what it requires from me is two things:
1) Energy level after I spend all day developing software at work, and
2) Getting the various `FHIRDate`, `FHIRTime`, `DateTime` working so that they actually make sense (currently it's nearly impossible to query FHIR objects in Realm based on date-time because I did a naive job of porting those from Pascal's SwiftFHIR).

My goal over the coming couple weeks is to
1) Get point #2 addressed, and
2) Just naively run the generator against the STU3 JSON specs and see what happens.

In theory, I believe, item #2 should work just peachy (according to James Agnew, anyways... we will see).

Apart from that I'll need to start gearing up for Swift 4 but I'm waiting for now.

TL;DR (though you likely already read my ramblings) - you've lit a fire under my ass and I'll see what I can do over the next couple weeks 😉 This has been an interesting exercise, Realm'izing FHIR.... it's good for some things, mediocre-to-bad for others, but overall I'd rather have this than not when working on a project.

**Comment by ryanbaldwin** [comment]
Hi Quyen,

I just wanted to follow up with the progress. Currently, sadly, it has not moved forward. I'm still trying to revamp the DateTime & Instant models so that they can be filtered/queried more naturally in Realm using standard Swift `Date` objects. I'm about half done on that. I'm hoping to finish that up within the next week, after which investigating STU3 is my primary goal. I will keep you posted here. 

My apologies for the delay.

Cheers!

**Comment by marceden** [comment]
Hi Ryan,

I can only guess what it’s like to maintain a rather niche-opensource project on a quickly evolving technology… But I have to ask: Do you have any plans updating FireKit to a more recent FHIR release or is this totally out of your scope?
Running the generator fails at the moment, even with STU3, and I'm figuring out whether this is a FireKit oder fhir-parser-issue…

However, would be great to hear from you, even if you've deciced to not longer maintain this project.


Cheers and kind regards,

Marc

**Comment by ryanbaldwin** [comment]
Hi Marc!

Well, sadly, I don't have much in the way of plans for updating to STU3. After 7 years I recently left Healthcare, and my ability to dogfood FireKit on a newer version is, alas, no longer existent. Most of my spare professional development time is getting caught up on context/know-how in a new domain with a new company.

Regarding the fhir-parser issue - Each release of FireKit is locked to a specific commit of fhir-parser in order to try and avoid parsing issues. It should, as far as I know, still work if you clone v2.402.07 (horrible naming, I apologize for that). Further more, the "newest" version of FireKit (which is what, 1.5 years old now? oye) doesn't play well with Realm 3+ (Realm 2.10 is the stable partner). 

In hindsight - while I love the functionality FireKit allowed - I think it's doing too much, in that it munges and conflates the layers. If I were to do this again, I would make it a proper bridging project which would be built on top of Smart-On-Fhir's offerings, and would provide a natural feeling API translating SOF's tech with an underlying Realm. 

My apologies for being a bad news bear, Marc. I should update the readme to reflect reality.

**Comment by marceden** [comment]
Nah, don’t feel bad about that. No worries, that’s how things are going and I totally got this. And thanks for your honest opinion on the state of FireKit with some distance now. I'll take that into account and I will think about how I'm gonna go on. Thanks again for your efforts and all of my best wishes for your future career. Kind regards from Germany, Marc
